md2moodle.py

```diff
@@ -184,16 +192,29 @@ def render_question(text, md_dir_path):
     text = re.sub(MULTI_LINE_CODE_PATTERN, replace_multi_line_code, text)
     text = re.sub(SINGLE_LINE_CODE_PATTERN, replace_single_line_code, text)
     text = re.sub(IMAGE_PATTERN, replace_image_wrapper(md_dir_path), text)
-    text = re.sub(DOUBLE_DOLLAR_LATEX_PATTERN, replace_latex, text)
+    text = re.sub(DOUBLE_DOLLAR_LATEX_PATTERN, replace_latex_double_dollars, text)
     text = re.sub(SINGLE_DOLLAR_LATEX_PATTERN, replace_latex, text)
     text = wrap_cdata( markdown_custom(text) )
     return text
 
 def markdown_custom(text):
     """Just calls markdown, but may be extended in the future."""
     #return markdown(text, extensions=['tables'])
-    return markdown(text, extensions=['nl2br'])
-    #return markdown(text)
+    # return markdown(text, extensions=['nl2br'])
+    return markdown(text)
+#TODO: Now, remove nl2br extension to allow writing math formulas using several lines but maybe it is preferable to remove new lines (replaced by spaces) before processing math formulas and put again the extension.
+
+
+def replace_latex_double_dollars(match):
+    # Take the part without the $$ at the beginning and at the end
+    code = match.group(1)
+
+    # Replace \\ by \\\\ in code
+    code = code.replace("\\", "\\\\")
```

Comment:
I found the problem here.
From what I was able to test, there must be a space character after the four backslashes, i.e.:

`code = code.replace("\\", "\\\\ ")`

If there isn't, my Moodle instance sometimes interprets something like `\\&` or `\\s.t.`, in that it tries to make sense of symbols `\&` or `\s`.

Reply:
Well done!!! 
I did not know that weird things like `\\&` could be interpreted by Moodle. A space should be definitely added after. 

Best,

Reply:
Can you update your pull request? Or do I merge it and then perform the fix?